pkg/networkservice/common/excludedprefixes/client.go

```diff
@@ -173,3 +211,38 @@ func validateIPs(ipContext *networkservice.IPContext, excludedPrefixes []string)
 
 	return nil
 }
+
+func getURL(request *networkservice.NetworkServiceRequest) *url.URL {
+	nsurl := &url.URL{}
+
+	nsurl.Host = request.GetConnection().GetNetworkService()
+	mechanism := request.GetConnection().GetMechanism()
+	if mechanism == nil && len(request.MechanismPreferences) > 0 {
+		mechanism = request.MechanismPreferences[0]
+	}
+
+	nsurl.Scheme = strings.ToLower(mechanism.GetType())
+	iface := mechanism.GetParameters()[common.InterfaceNameKey]
+	if iface != "" {
+		nsurl.Path = "/" + iface
+	}
+	query := nsurl.Query()
+	for k, v := range request.GetConnection().GetLabels() {
+		query.Add(k, v)
+	}
+	nsurl.RawQuery = query.Encode()
+
+	return nsurl
+}
+
+func checkAwarenessGroups(nsurl *url.URL, awarenessGroups [][]*url.URL) (isInGroup bool, i int) {
```

Comment:
Can we check `isInGroup` if index >= 0?


```suggestion
func checkAwarenessGroups(nsurl *url.URL, awarenessGroups [][]*url.URL) int {
```